### PR TITLE
aws-lc: init at 1.37.0

### PR DESCRIPTION
**Pull request by theoparis**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

I have marked this as a draft because `pkgsStatic.aws-lc` fails to build and I'm yet to figure out the "no such file or directory" errors.

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/aw/aws-lc/package.nix [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      package = aws-lc;
+      moduleNames = [
+        "libcrypto"
+        "libssl"
+        "openssl"
+      ];
+    };
+  };
+
+  meta = with lib; {
```
Comment (RossComputerGuy) [inline pkgs/by-name/aw/aws-lc/package.nix:82]: Don't use `with lib`

File: pkgs/by-name/aw/aws-lc/package.nix [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  stdenv,
+  buildGoModule,
+  cmake,
+  fetchFromGitHub,
+  ninja,
+  testers,
+  aws-lc,
+}:
+
+buildGoModule rec {
```
Comment (RossComputerGuy) [inline pkgs/by-name/aw/aws-lc/package.nix:12]: Don't use `rec`, use `finalAttrs:`

File: pkgs/by-name/aw/aws-lc/package.nix [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  pname = "aws-lc";
+  version = "1.33.0";
+
+  src = fetchFromGitHub {
+    owner = "aws";
+    repo = "aws-lc";
+    rev = "v${version}";
```
Comment (RossComputerGuy) [inline pkgs/by-name/aw/aws-lc/package.nix:19]: Can reference `finalAttrs.version`

**Review by RossComputerGuy — CHANGES_REQUESTED** [review]

**Comment by RossComputerGuy** [comment]
Result of `nixpkgs-review pr 333578` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>aws-lc</li>
    <li>aws-lc.bin</li>
    <li>aws-lc.dev</li>
  </ul>
</details>

File: pkgs/by-name/aw/aws-lc/package.nix (hunk 2 above)
Comment (RossComputerGuy) [reply pkgs/by-name/aw/aws-lc/package.nix:12]: Looks like `buildGoModule` doesn't support `finalAttrs`.

**Comment by theoparis** [comment]
Hmm, there seems to be failing tests on darwin platforms and I'm not sure what's going on.
> `/nix/store/hmchx96ysd0h19p7kx0w12pljkbp30j3-Libsystem-1238.60.2/include/CommonCrypto/CommonRandom.h:35:9: error: unknown type name 'CCCryptorStatus'`

**Comment by theoparis** [comment]
Unfortunately I don't know why aarch64-darwin is stuck on being queued.

**Comment by theoparis** [comment]
Sorry I messed up, new PR is here #356628

**Pull request closed without merging**